### ISSUE TITLE
SC2016 false positive: awk expression with `$` fails if prepended by `command -p`

### ISSUE THREAD

**Issue by LucasLarson** [koalaman/shellcheck#2773]
#### For bugs
- Rule Id: SC2016
- My shellcheck version: 0.9.0
- [x] [The rule's wiki page](https://shellcheck.net/wiki/SC2016) does not already cover this
- [x] I tried on [shellcheck.net](https://www.shellcheck.net/) and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] [shellcheck.net](https://www.shellcheck.net/) (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through [koalaman/shellcheck/issues](https://github.com/koalaman/shellcheck/issues) and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

# these cause SC2016 false positives:
command -p -- awk -- '! seen[$0]++ {print}' "$@"
command -p -- awk -- '!seen[$0]++ {print}' "$@"
command -p -- awk '! seen[$0]++ {print}' "$@"
command -p -- awk '!seen[$0]++ {print}' "$@"
command -p awk -- '! seen[$0]++ {print}' "$@"
command -p awk -- '!seen[$0]++ {print}' "$@"
command -p awk '! seen[$0]++ {print}' "$@"
command -p awk '!seen[$0]++ {print}' "$@"

# these pass shellcheck:
command awk -- '! seen[$0]++ {print}' "$@"
command awk '! seen[$0]++ {print}' "$@"
command awk '!seen[$0]++ {print}' "$@"
awk -- '! seen[$0]++ {print}' "$@"
awk '! seen[$0]++ {print}' "$@"
awk '!seen[$0]++ {print}' "$@"
```

#### Here's what shellcheck currently says:
```console
seener:4:22: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:5:22: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:6:19: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:7:19: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:8:19: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:9:19: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:10:16: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
seener:11:16: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
```

#### Here's what I wanted or expected to see:
[Nothing]

**Comment by LucasLarson** [comment]
Closing as a probable duplicate of
- https://github.com/koalaman/shellcheck/issues/382
- https://github.com/koalaman/shellcheck/issues/501
- https://github.com/koalaman/shellcheck/issues/1059
- https://github.com/koalaman/shellcheck/issues/1293
- https://github.com/koalaman/shellcheck/issues/1536
- https://github.com/koalaman/shellcheck/issues/1894
- https://github.com/koalaman/shellcheck/issues/1949
- https://github.com/koalaman/shellcheck/issues/1955
- https://github.com/koalaman/shellcheck/issues/2693
